framework/fp_vectors/Floats.h

```diff
@@ -273,16 +352,60 @@ typedef union {
 static_assert(sizeof(double) < sizeof(long double), "Compiler does not support long double");
 static_assert(sizeof(Float80) == sizeof(long double), "Float80 structure is not of the correct size");
 
-Float16 new_float16(unsigned sign, unsigned exponent, unsigned mantissa);
+Float16 new_float16(uint32_t sign, uint32_t exponent, uint32_t mantissa);
+BFloat16 new_bfloat16(uint32_t sign, uint32_t exponent, uint32_t mantissa);
 Float32 new_float32(uint32_t sign, uint32_t exponent, uint32_t mantissa);
 Float64 new_float64(uint32_t sign, uint32_t exponent, uint64_t mantissa);
 Float80 new_float80(uint32_t sign, uint32_t exponent, uint32_t jbit, uint64_t mantissa);
 
 Float16 new_random_float16();
+BFloat16 new_random_bfloat16();
 Float32 new_random_float32();
 Float64 new_random_float64();
 Float80 new_random_float80();
 
+// __builtins
+#define IS_NEGATIVE(v) \
+    _Generic((v),\
+        Float16: Float16_is_negative,\
+        BFloat16: BFloat16_is_negative\
+    )(v)
+#define IS_ZERO(v) \
+    _Generic((v),\
+        Float16: Float16_is_zero,\
+        BFloat16: BFloat16_is_zero\
+    )(v)
+#define IS_DENORMAL(v) \
+    _Generic((v),\
+        Float16: Float16_is_denormal,\
+        BFloat16: BFloat16_is_denormal\
+    )(v)
+#define IS_INF(v) \
+    _Generic((v),\
+        Float16: Float16_is_inf,\
+        BFloat16: BFloat16_is_inf\
+    )(v)
+#define IS_NAN(v) \
+    _Generic((v),\
+        Float16: Float16_is_nan,\
+        BFloat16: BFloat16_is_nan\
+    )(v)
+#define IS_SNAN(v) \
+    _Generic((v),\
+        Float16: Float16_is_snan,\
+        BFloat16: BFloat16_is_snan\
+    )(v)
+#define IS_QNAN(v) \
+    _Generic((v),\
+        Float16: Float16_is_qnan,\
+        BFloat16: BFloat16_is_qnan\
+    )(v)
+#define GET_NAN_PAYLOAD(v) \
+    _Generic((v),\
+        Float16: Float16_get_nan_payload,\
+        BFloat16: BFloat16_get_nan_payload\
+    )(v)
```

Comment:
Suggestion: these can be extended for the other three floating point types. Later.